src/dune_rules/melange/melange_binary.ml

```diff
@@ -13,20 +13,27 @@ let where =
       @@ Process.run_capture_line ~display:Quiet Process.Strict bin
            [ "--where" ]
     in
-    Path.of_string where
+    where
   in
   let memo =
-    Memo.create "melange-where" ~input:(module Path) ~cutoff:Path.equal impl
+    Memo.create "melange-where" ~input:(module Path) ~cutoff:String.equal impl
   in
   fun sctx ~loc ~dir ->
     let open Memo.O in
     let* env = Super_context.env_node sctx ~dir >>= Env_node.external_env in
-    match Env.get env "MELANGELIB" with
-    | Some p -> Memo.return (Some (Path.of_string p))
-    | None -> (
-      let* melc = melc sctx ~loc ~dir in
-      match melc with
-      | Error _ -> Memo.return None
-      | Ok melc ->
-        let+ res = Memo.exec memo melc in
-        Some res)
+    let+ melange_dirs =
+      match Env.get env "MELANGELIB" with
+      | Some p -> Memo.return (Some p)
+      | None -> (
+        let* melc = melc sctx ~loc ~dir in
+        match melc with
+        | Error _ -> Memo.return None
+        | Ok melc ->
+          let+ res = Memo.exec memo melc in
+          Some res)
+    in
+    Option.map
+      ~f:(fun dirs ->
+        String.split ~on:Bin.path_sep dirs |> List.map ~f:Path.of_string)
```

Comment:
If this is a relative path, is it clear what it should be relative to?

Reply:
relative to the `melc` executable

Reply:
perhaps I should clarify: the result of `melc --where` is guaranteed to be an absolute path. So I believe dune doesn't have to worry about it being a relative path.

Reply:
If it can't be a relative path, perhaps the code shouldn't allow it. E.g. `Path.External.of_string`.

Reply:
Changed to a `Path.External.t` in [44cf928](https://github.com/ocaml/dune/pull/7176/commits/44cf92884ae9beca062597bef74ba09110ca429b)